solr/core/src/java/org/apache/solr/cloud/api/collections/CategoryRoutedAlias.java

```diff
@@ -125,13 +146,13 @@ String buildCollectionNameFromValue(String value) {
 
   @Override
   public String createCollectionsIfRequired(AddUpdateCommand cmd) {
+    assert !getCollectionList(this.parsedAliases).contains(buildCollectionNameFromValue(String.valueOf(cmd.getSolrInputDocument().getFieldValue(getRouteField()))));
```

Comment:
This assert is not correct... the method is createCollections--if--required. meaning it might not be required (because it might already exist) I'll remove this.

Reply:
Oh sorry I missed that subtle nuance.